playback/manifests.go

```diff
@@ -5,37 +5,51 @@ import (
 	"fmt"
 	"io"
 	"net/url"
-	"path"
+	"strings"
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/grafov/m3u8"
 	"github.com/livepeer/catalyst-api/clients"
 	"github.com/livepeer/catalyst-api/config"
 	caterrs "github.com/livepeer/catalyst-api/errors"
+	"github.com/livepeer/go-tools/drivers"
 )
 
-const ManifestKeyParam = "tkn"
+const KeyParam = "tkn"
 
-type PlaybackRequest struct {
+type Request struct {
 	RequestID  string
 	PlaybackID string
 	File       string
 	AccessKey  string
 }
 
-func Manifest(req PlaybackRequest) (io.Reader, error) {
-	if req.AccessKey == "" {
-		return nil, fmt.Errorf("invalid request: %w", caterrs.EmptyAccessKeyError)
-	}
+type Response struct {
+	Body        io.ReadCloser
+	ContentType string
+}
 
-	reader, err := osFetch(req.PlaybackID, req.File)
+func Handle(req Request) (*Response, error) {
+	f, err := osFetch(req.PlaybackID, req.File)
 	if err != nil {
 		return nil, err
 	}
-	defer reader.Close()
 
-	p, listType, err := m3u8.DecodeFrom(reader, true)
+	if !strings.HasSuffix(req.File, "m3u8") && f.ContentType != "application/x-mpegurl" {
```

Comment:
Perhaps skip the content-type check. We don't want to risk trying to parse other kinds of files as manifests (e.g. metadata or src files) and failing. Also would add `.` prefix to m3u8, wdyt?
```suggestion
	if !strings.HasSuffix(req.File, ".m3u8") {
```